modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/HttpClient.scala

```diff
@@ -13,28 +13,80 @@
 package com.snowplowanalytics.snowplow.enrich.common.utils
 
 import scala.util.control.NonFatal
-
-import cats.Id
+import cats.{Applicative, Id}
 import cats.effect.Sync
-import cats.syntax.either._
+import cats.implicits._
+import fs2.Stream
+import org.http4s.client.{Client => Http4sClient}
+import org.http4s.{EmptyBody, EntityBody, Method, Request, Uri}
 import scalaj.http._
 
 trait HttpClient[F[_]] {
-  def getResponse(request: HttpRequest): F[Either[Throwable, String]]
+  def getResponse(
+    uri: String,
+    authUser: Option[String],
+    authPassword: Option[String],
+    body: Option[String],
+    method: String,
+    connectionTimeout: Option[Long],
+    readTimeout: Option[Long]
+  ): F[Either[Throwable, String]]
 }
 
 object HttpClient {
   def apply[F[_]](implicit ev: HttpClient[F]): HttpClient[F] = ev
 
-  implicit def syncHttpClient[F[_]: Sync]: HttpClient[F] =
+  implicit def syncHttpClient[F[_]: Sync](implicit http4sClient: Http4sClient[F]): HttpClient[F] =
     new HttpClient[F] {
-      override def getResponse(request: HttpRequest): F[Either[Throwable, String]] =
-        Sync[F].delay(getBody(request))
+      override def getResponse(
+        uri: String,
+        authUser: Option[String],
+        authPassword: Option[String],
+        body: Option[String],
+        method: String,
+        connectionTimeout: Option[Long],
+        readTimeout: Option[Long]
+      ): F[Either[Throwable, String]] = {
+        val request = Request[F](
+          uri = Uri.unsafeFromString(uri),
```

Comment:
I know this _should_ be safe, but I don't like seeing things like `unsafeFromString` in the code!  Can you think of any ways to avoid it?

Reply:
good point, I replaced that with a safer option